### PR TITLE
[#3009] improve(web): improve the user experience of refreshing the oauth token

### PR DESCRIPTION
**Pull request by ch3yne**
### What changes were proposed in this pull request?

Improve the user experience of refreshing the oauth token.

### Why are the changes needed?

Fix: #3009

### Does this PR introduce _any_ user-facing change?

When the user doesn't interact with the interface for 30 minutes, they will be brought back to the login page.

### How was this patch tested?

Local

### CODE REVIEW

File: web/src/lib/provider/session.js [hunk 1]
```diff
@@ -35,6 +37,14 @@ const AuthProvider = ({ children }) => {
   const searchParams = useSearchParams()
   const paramsSize = [...searchParams.keys()].length
 
+  const isIdle = useIdle(18e5) // 30 minutes
```
Comment (LauraXia123) [inline web/src/lib/provider/session.js:40]: What's scope of action will broke the idle? And the `initialState` is settled, Can we use the configurable value.

File: web/src/lib/store/auth/index.js [hunk 2]
```diff
@@ -61,10 +61,11 @@ export const loginAction = createAsyncThunk('auth/loginAction', async ({ params,
     throw new Error(err)
   }
 
-  const { access_token, expires_in } = res
+  const { access_token, expires_in } = res // `expires_in ` is in seconds, default is 499
 
   localStorage.setItem('accessToken', access_token)
   localStorage.setItem('expiredIn', expires_in)
+  localStorage.setItem('isIdle', false)
```
Comment (LauraXia123) [inline web/src/lib/store/auth/index.js:68]: only after login idle is false, any other event action?

File: web/src/lib/provider/session.js (hunk 1 above)
Comment (ch3yne) [reply web/src/lib/provider/session.js:40]: Please check [useIdle(react-use)](https://github.com/streamich/react-use/blob/master/src/useIdle.ts).
The default events list is 'mousemove', 'mousedown', 'resize', 'keydown', 'touchstart', 'wheel'.

File: web/src/lib/store/auth/index.js (hunk 2 above)
Comment (ch3yne) [reply web/src/lib/store/auth/index.js:68]: [web/src/lib/provider/session.js](https://github.com/datastrato/gravitino/pull/3512/files#diff-6b423168fab9ac6f071121b402c464795d674cfd97d2aab29da8794c3becf093)#L42-L46
After exceeding the set time period, isIdle will be set to true.

**Comment by LauraXia123** [comment]
LGTM

**Review by LauraXia123 — APPROVED** [review]
LGTM

**Pull request merged**